tests/lib/config/flat-config-schema.js

```diff
@@ -100,22 +100,28 @@ describe("merge", () => {
         assert.notStrictEqual(result, second);
     });
 
-    it("overwrites a value in the first object with null in the second one", () => {
+    it("throws an error if a value in a property is overriden with null", () => {
         const first = { foo: { bar: "baz" } };
         const second = { foo: null };
-        const result = merge(first, second);
 
-        assert.deepStrictEqual(result, second);
-        assert.notStrictEqual(result, second);
+        assert.throws(() => merge(first, second), TypeError);
     });
 
-    it("overwrites a value in the first object with a primitive in the second one", () => {
+    it("does not override a value in a property with a primitive", () => {
         const first = { foo: { bar: "baz" } };
         const second = { foo: 42 };
         const result = merge(first, second);
 
-        assert.deepStrictEqual(result, second);
-        assert.notStrictEqual(result, second);
+        assert.deepStrictEqual(result, first);
+        assert.notStrictEqual(result, first);
+    });
+
+    it("merges an object in a property with a string", () => {
```

Comment:
This is the current behavior. It's because of the destructuring here:

https://github.com/eslint/eslint/blob/5de9637fc925729a83d5a5e9e868a41792a184e3/lib/config/flat-config-schema.js#L86-L89

A primitive like `"qux"` is converted into an object during destructuring i.e. `Object("qux")`, which is similar to `{ "q": 0, "u": 1, "x": 2 }`.

A number like `42` will be also converted into an object when destructured, but since `Object(42)` has no enumerable properties, the effect won't be visible, and it will behave like an empty object.

Reply:
Gotcha. Thanks for adding tests to make this clear. 👍